### PR TITLE
IOTSSL-1879 buffer overflow in server key exchange parsing

### PR DESCRIPTION
**Pull request by k-stachowiak**
## Description
This PR prevents two buffer overflow errors from happening in the ssl_parse_server_key_exchange function.

## Status
**READY**

## Requires Backporting
YES
This fix should most probably be backported to all supported branches as the vulnerabilities aren't version speciffic.

## Migrations
NO

## Additional comments
None

## Todos
- [ ] Tests maybe?

## Steps to test or reproduce
None prepared

### CODE REVIEW

**Comment by k-stachowiak** [comment]
I have just noticed the verification build failure. I will investigate it.

**Comment by gilles-peskine-arm** [comment]
CI passed except for the FreeBSD timing self-test which is a known issue.

**Comment by gilles-peskine-arm** [comment]
Can you please add a ChangeLog entry, and make backports to 2.1 and 2.7?

File: ChangeLog [hunk 1]
```diff
@@ -13,6 +13,8 @@ Default behavior changes
      Applied Sciences).
 
 Security
+   * Fix buffer overread vulnerability in ssl_parse_server_key_exchange
+     function.
```
Comment (gilles-peskine-arm) [inline ChangeLog]: Same remarks as https://github.com/ARMmbed/mbedtls/pull/1440/files/5224a7544c95552553e2e6be0b4a789956a6464e..999a3bce6e19e3300828584648b20d2784bf882f#r174206943 and https://github.com/ARMmbed/mbedtls/pull/1440/files/5224a7544c95552553e2e6be0b4a789956a6464e..999a3bce6e19e3300828584648b20d2784bf882f#r174207812

File: ChangeLog [hunk 2]
```diff
@@ -17,6 +17,13 @@ Security
      implementation allowed an offline 2^80 brute force attack on the
      HMAC key of a single, uninterrupted connection (with no
      resumption of the session).
+   * Fix buffer overread vulnerability in ssl_parse_server_key_exchange
+     function. Previous implementation could in theory lead to a segmentation
+	 fault upon reading invalid input.
```
Comment (gilles-peskine-arm) [inline ChangeLog]: As https://github.com/ARMmbed/mbedtls/pull/1440/files#r174403294:

> Fix a buffer overread in ssl_parse_server_key_exchange() that could cause a crash on invalid input.

File: ChangeLog [hunk 3]
```diff
@@ -17,6 +17,13 @@ Security
...
+   * Fix possible arithmetic overflow on bounds check in the
+     ssl_parse_server_key_exchange function. The overflow could potentially lead
+	 to incorrect assertion of a buffer bounds and a termination of the key
+	 exchange even upon receiving correct data.
```
Comment (gilles-peskine-arm) [inline ChangeLog]: As https://github.com/ARMmbed/mbedtls/pull/1440/files#r174403294:

> Fix a possible arithmetic overflow in ssl_parse_server_key_exchange() that could cause a key exchange to fail on valid data.

**Comment by gilles-peskine-arm** [comment]
retest

**Comment by gilles-peskine-arm** [comment]
retest

**Pull request merged**